src/device_trezor/device_trezor_base.cpp

```diff
@@ -365,15 +365,14 @@ namespace trezor {
     void device_trezor_base::device_state_initialize_unsafe()
     {
       require_connected();
-      std::string tmp_session_id;
       auto initMsg = std::make_shared<messages::management::Initialize>();
       const auto data_cleaner = epee::misc_utils::create_scope_leave_handler([&]() {
-        memwipe(&tmp_session_id[0], tmp_session_id.size());
+        if (initMsg->has_session_id())
+          memwipe(&(*initMsg->mutable_session_id())[0], initMsg->mutable_session_id()->size());
```

Comment:
Would like to hear @vtnerd’s source for "The m_body[0] expression is only valid if !m_body.empty()."

Reply:
My fault, the [the C++ specification changed](https://en.cppreference.com/w/cpp/string/basic_string/operator_at) since C++11. The expression `m_body[0]` is valid when `m_body.size() == 0` but the value will be `0` and the behavior is undefined if the value is changed. So this expression and the change in the PR are unnecessary since its always setting the value to `0`. The prior C++ spec was stricter but more obvious.

Reply:
It's also not inherently wrong to check, because that still meets the requirement for not changing the `NULL` character.